### PR TITLE
Add support for QuantaMesh T1048-LY4R

### PR DESCRIPTION
**Pull request by audihsu-qci**
QuantaMesh T1048-LY4R Specification
 - 48x10/100/1000BASE-T and 4x1/10GbE SFP+ ports
 - 1 RJ-45 out-of-band management port (10/100/1000M)
 - 1 1GbE SFP out-of-band management port
 - 1 RJ-45 console port
 - 1 4GB DDR3 SDRAM
 - 1 8GB microSD card
 - 1 USB 2.0 port
 - BMC ipmitool support

For further information, please take a look at machine/quanta/quanta_ly4r/INSTALL

### CODE REVIEW

File: machine/quanta/quanta_ly4r/INSTALL [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+? UEFI: NIC1 IPv6 Intel(R) Ethernet Connection      ?/-: Change Opt.         ?
+? I354                                              ｳF1: Help for more Keys   ?
+? Generic Ultra HS-COMBO                            ｳF8: Previous Values      ?
+? AMI Virtual CDROM0 1.00                           ｳF9: Optimized Defaults   ?
+? ONIE: Open Network Install Environment            ｳF10: Save & Reset        ?
+?                                                   ｳESC: Exit                ?
+ﾀﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾁﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾄﾙ
+        Version 2.17.1245. Copyright (C) 2017 American Megatrends, Inc.
+                                                                             AB
+
```
Comment (cbrune) [inline machine/quanta/quanta_ly4r/INSTALL]: I really appreciate the BIOS screenshots.  This is very helpful.  Thanks!

File: machine/quanta/quanta_ly4r/installer.conf [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+# Local Variables:
+# mode: shell-script
+# eval: (sh-set-shell "/bin/sh" t nil)
+# End:
+
+
+#
+# Sample usage for pre/post install hooks
+#
+pre_install()
```
Comment (cbrune) [inline machine/quanta/quanta_ly4r/installer.conf]: if you don't need this function just leave it out.

File: machine/quanta/quanta_ly4r/kernel/fix-i2c-ismt-write-error.patch [hunk 3]
```diff
@@ -0,0 +1,33 @@
+fix i2c-ismt write error
+
+diff --git a/drivers/i2c/busses/i2c-ismt.c b/drivers/i2c/busses/i2c-ismt.c
+index 6e932d1..7dd87b8 100644
+--- a/drivers/i2c/busses/i2c-ismt.c
++++ b/drivers/i2c/busses/i2c-ismt.c
+@@ -675,6 +675,23 @@ static irqreturn_t ismt_do_interrupt(int vec, void *data)
+  */
+ static irqreturn_t ismt_do_msi_interrupt(int vec, void *data)
```
Comment (cbrune) [inline machine/quanta/quanta_ly4r/kernel/fix-i2c-ismt-write-error.patch]: what problem is this patch fixing?

File: machine/quanta/quanta_ly4r/kernel/fix-i2c-ismt-write-error.patch [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ {
++	u32 val;
++	struct ismt_priv *priv = data;
++
++	val = readl(priv->smba + ISMT_MSTR_MSTS);
++
++	while (val & (ISMT_MSTS_IP))
++	{
++		val = readl(priv->smba + ISMT_MSTR_MSTS);
++	}
```
Comment (cbrune) [inline machine/quanta/quanta_ly4r/kernel/fix-i2c-ismt-write-error.patch]: The usual idiom here is use a `do { .... } while();` construct.  Then you do not need the addition readl() done before the while().
```
do {
    val = readl(priv->smba + ISMT_MSTR_MSTS);
} while (val & (ISMT_MSTS_IP));
```

File: machine/quanta/quanta_ly4r/rootconf/sysroot-lib-onie/sysinfo-platform [hunk 5]
```diff
@@ -0,0 +1,21 @@
+
+#  Copyright (C) 2014 Curt Brune <curt@cumulusnetworks.com>
+#
+#  SPDX-License-Identifier:     GPL-2.0
+
+get_serial_num()
+{
+    [ -r /usr/bin/onie-syseeprom ] && /usr/bin/onie-syseeprom -g 0x23
```
Comment (cbrune) [inline machine/quanta/quanta_ly4r/rootconf/sysroot-lib-onie/sysinfo-platform]: The test should be `-x`, which makes sure the file is readable and executable.

File: machine/quanta/quanta_ly4r/kernel/fix-i2c-ismt-write-error.patch (hunk 3 above)
Comment (audihsu-qci) [inline machine/quanta/quanta_ly4r/kernel/fix-i2c-ismt-write-error.patch]: We use i2c-ismt as /dev/i2c-1, which we put board info eeprom here.
We found the i2cset command did not complete normally, Therefore we added the while loop to wait the in progress flag complete, then clear the interrupt flags.

**Pull request closed without merging**